### PR TITLE
ref: pre v2.15 cleanup

### PR DESCRIPTION
**Pull request by elchead**
### Context
This cleans up open TODOs before starting the new release.

### Proposed change(s)

-

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Add labels (e.g., for changelog category)
- [ ] Is PR title adequate for changelog?
- [ ] Link to Milestone

### CODE REVIEW

File: internal/constellation/helm/actionfactory.go [hunk 1]
```diff
@@ -120,11 +120,7 @@ func (a actionFactory) appendNewAction(
 		} else {
 			// This may break for external chart dependencies if we decide to upgrade more than one minor version at a time.
 			if err := newVersion.IsUpgradeTo(currentVersion); err != nil {
-				// TODO(3u13r): Remove when Constellation v2.14 is released.
-				// We need to ignore that we jump from Cilium v1.12 to v1.15-pre. We have verified that this works.
-				if !(errors.Is(err, compatibility.ErrMinorDrift) && (release.releaseName == "cilium" || release.releaseName == "cert-manager")) {
-					return fmt.Errorf("invalid upgrade for %s: %w", release.releaseName, err)
-				}
```
Comment (3u13r) [inline internal/constellation/helm/actionfactory.go:127]: I think we want to allow larger jumps for the cert-manager and cilium in the future. Therefore, I'd keep the code and adjust the comment.

File: internal/constellation/kubecmd/kubecmd.go [hunk 2]
```diff
@@ -104,11 +103,6 @@ func (k *KubeCmd) UpgradeNodeImage(ctx context.Context, imageVersion semver.Semv
 		return fmt.Errorf("updating image version: %w", err)
 	}
 
-	// TODO(3u13r): remove `reconcileKubeadmConfigMap` after v2.14.0 has been released.
```
Comment (3u13r) [inline internal/constellation/kubecmd/kubecmd.go:107]: Note that there are also a lot of test code which must be deleted.

**Review by 3u13r — CHANGES_REQUESTED** [review]

**Review by msanft — APPROVED** [review]
"My" change in the Terraform variables LGTM. Thanks for taking care of this!

File: operators/constellation-node-operator/internal/upgrade/upgrade.go [hunk 3]
```diff
@@ -44,22 +44,8 @@ func (c *Client) Upgrade(ctx context.Context, kubernetesComponents components.Co
...
-	// KubernetesComponents proto field. Therefore, we pass the kubeadm
-	// component twice: once via KubeadmUrl/KubeadmHash, once as part of the
-	// kubernetesComponents argument.
-	kubeadm, err := kubernetesComponents.GetKubeadmComponent()
-	if err != nil {
-		return fmt.Errorf("expected a kubeadm Component: %w", err)
-	}
 	protoClient := upgradeproto.NewUpdateClient(conn)
 	_, err = protoClient.ExecuteUpdate(ctx, &upgradeproto.ExecuteUpdateRequest{
-		// TODO(burgerdev): remove these fields after releasing 2.14.
```
Comment (burgerdev) [inline operators/constellation-node-operator/internal/upgrade/upgrade.go:58]: There is a corresponding simplification opportunity in `upgrade-agent/internal/server/server.go`, and we could remove the kubeadm* fields from the proto message.

File: upgrade-agent/upgradeproto/upgrade.proto [hunk 4]
```diff
@@ -11,8 +11,8 @@ service Update {
 }
 
 message ExecuteUpdateRequest {
-  string kubeadm_url = 1;
-  string kubeadm_hash = 2;
+  reserved 1;
+  reserved 2;
```
Comment (burgerdev) [inline upgrade-agent/upgradeproto/upgrade.proto:15]: Nit: field names should also be reserved, https://protobuf.dev/programming-guides/proto3/#fieldreserved.
```suggestion
  reserved 1, 2;
  reserved "kubeadm_url", "kubeadm_hash";
```

**Review by burgerdev — APPROVED** [review]

File: internal/constellation/helm/actionfactory.go [hunk 5]
```diff
@@ -120,8 +120,7 @@ func (a actionFactory) appendNewAction(
 		} else {
 			// This may break for external chart dependencies if we decide to upgrade more than one minor version at a time.
 			if err := newVersion.IsUpgradeTo(currentVersion); err != nil {
-				// TODO(3u13r): Remove when Constellation v2.14 is released.
-				// We need to ignore that we jump from Cilium v1.12 to v1.15-pre. We have verified that this works.
+				// Allow bigger Cilium version jumps.
```
Comment (3u13r) [inline internal/constellation/helm/actionfactory.go:123]:
```suggestion
				// Allow bigger Cilium and Cert-Manager version jumps.
```

File: internal/constellation/kubecmd/kubecmd_test.go [hunk 6]
```diff
@@ -57,38 +57,6 @@ func TestUpgradeNodeImage(t *testing.T) {
 			constants.ClusterConfigurationKey: string(clusterConfBytes),
 		},
 	}
-
-	clusterConfWithKonnectivity := kubeadmv1beta3.ClusterConfiguration{
```
Comment (3u13r) [inline internal/constellation/kubecmd/kubecmd_test.go:61]: Please have a look at the commit that introduced those tests and clean up everything added there.

**Review by 3u13r — CHANGES_REQUESTED** [review]

**Review by 3u13r — APPROVED** [review]
LGTM

**Pull request merged**